src/VirtualClient/VirtualClient.Actions/CoreMark/CoreMarkExecutor.cs

```diff
@@ -108,17 +108,22 @@ private async Task ExecuteWorkloadAsync(string pathToExe, string commandLineArgu
             {
                 if (!cancellationToken.IsCancellationRequested)
                 {
+                    process.ProcessDetails.ToolSet = "CoreMark";
                     if (process.IsErrored())
                     {
-                        await this.LogProcessDetailsAsync(process, telemetryContext, "CoreMark", logToFile: true);
+                        await this.LogProcessDetailsAsync(process.ProcessDetails, telemetryContext, logToFile: true);
```

Comment:
I would like to revisit this implementation. I think that we can probably minimize the number of changes here by placing the point at which we reference the ProcessDetails abstraction within an extension method so we can still use the IProcessProxy as is. We will add in an extension method to match for the ISshclientProxy that takes in whatever parameters it needs to take in. The 3rd extension method would then take in a ProcessDetails object and would be called by the other 2 here underneath.